File: final.py
// mapper.py

import sys
for line in sys.stdin:
line=line.strip()
words=line.split()
for word in words:
print('%s\t%s'%(word,1))


// reducer.py

from operator import itemgetter
import sys
current_word=None
current_count=0
word=None
for line in sys.stdin:
line=line.strip()
word,count=line.split('%t',1)

try:
	count=int(count)
	except ValueError:
	continue
		if current_word==word:
		   current_count+=count
		   else:
			if current_word:
 			print('%s\t%s'%(current_word,current_count))
			current_word=word
			current_count=count
				if current_word==word:
				print('%s\t%s'%(current_word,current_count))
	


 

File: final.java
// mapper.java

import java.io.IOException;

import org.apache.hadoop.io.IntWritable;
import org.apache.hadoop.io.LongWritable;
import org.apache.hadoop.io.Text;
import org.apache.hadoop.mapreduce.Mapper;

public class WordCounterMap extends Mapper <LongWritable,Text,Text,IntWritable>
{

	protected void map(LongWritable key, Text value,Context context)
	throws IOException, InterruptedException
	{
		String [] words=value.toString().split(",");
		for(String word:words)
		context.write(new Text(word),new IntWritable(1))
		
	}


}

// reduce.java

import java.io.IOException;
import org.apache.hadoop.io.LongWritable;
import org.apache.hadoop.io.IntWritable;
import org.apache.hadoop.io.Text;
import org.apache.hadoop.mapreduce.Reducer;

public class WordcounterRed extends Reducer <Text,IntWritable,Text,LongWritable>
{
	protected void red(Text word, Iterable<IntWritable> values,Iterable<LongWriotable> keys,Context context)
	throws IOExcedption, InterruptedException
	{
		Integer count=0;
		for(IntWritable val: values)
		{
			count+=val.get();	
		}
		for(LongWritable key: keys)
		{
			count+=key.get();
		}
			context.write(word,new IntWritable(count),Longwritable(count));
}
}


	}


}
